src/components/VideoPlayer/BaseVideoPlayer.tsx

```diff
@@ -99,34 +103,35 @@ function BaseVideoPlayer({
     );
 
     const handlePlaybackStatusUpdate = useCallback(
-        (e) => {
-            const isVideoPlaying = e.isPlaying || false;
-            const currentDuration = e.durationMillis || videoDuration * 1000;
-            const currentPositon = e.positionMillis || 0;
+        (status: AVPlaybackStatus) => {
+            const isVideoPlaying = 'isPlaying' in status ? status.isPlaying : false;
```

Comment:
Let's add an early return here, this way the logic can be much simpler. Wdyt @smelaa?